### PR TITLE
Varda child full reset

### PR DESCRIPTION
**Pull request by reynders**

### CODE REVIEW

**Comment by hruoho** [comment]
Do you think we should delete the child altogether?

**Review by hruoho — APPROVED** [review]

File: service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt [hunk 1]
```diff
@@ -81,6 +105,77 @@ class VardaUpdateServiceV2(
     }
 }
 
+fun deleteChildDataFromVardaAndDb(db: Database.Connection, vardaClient: VardaClient, evakaChildId: UUID): Boolean {
+    logger.info("VardaUpdate: resetting all varda data for evaka child $evakaChildId")
+
+    val vardaChildIds = getVardaChildIdsByUUID(db, evakaChildId)
```
Comment (reynders) [inline service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt:111]: getVardaChildIdsByUUID -> getVardaChildIdsByEvakaChildId

File: service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt [hunk 2]
```diff
@@ -81,6 +105,77 @@ class VardaUpdateServiceV2(
...
+                }
+            }
+
+            db.transaction {
+                it.deleteVardaServiceNeedByEvakaChildId(evakaChildId)
+            }
+
+            return true
+        } catch (e: Exception) {
+            logger.info("VardaUpdate: couldn't reset varda child: ${e.localizedMessage}")
```
Comment (reynders) [inline service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt:154]: Log vardaChildId for debugging

File: service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt [hunk 3]
```diff
@@ -81,6 +105,77 @@ class VardaUpdateServiceV2(
...
+            val decisionIds = vardaClient.getDecisionsByChild(vardaChildId)
+            logger.info { "VardaUpdate: found decision ids for child $vardaChildId: $decisionIds" }
+
+            val placementIds = decisionIds.flatMap { vardaClient.getPlacementsByDecision(it) }
+            logger.info { "VardaUpdate: found placement ids for child $vardaChildId: $placementIds" }
+
+            val feeIds = placementIds.flatMap { vardaClient.getFeeDataByChild(it) }
+            logger.info { "VardaUpdate: found fee data ids for child $vardaChildId: $feeIds" }
+
+            logger.info { "VardaUpdate: Deleting ${feeIds.size} fee data records" }
```
Comment (reynders) [inline service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt:124]: Log the vardaChildId also

File: service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt [hunk 4]
```diff
@@ -81,6 +105,77 @@ class VardaUpdateServiceV2(
...
+            feeIds.forEach { vardaId ->
+                if (vardaClient.deleteFeeDataV2(vardaId)) {
+                    logger.info { "VardaUpdate: Deleting fee data from db by id $vardaId" }
+                    db.transaction { deleteVardaFeeData(it, vardaId) }
+                }
+            }
+
+            logger.info { "VardaUpdate: Deleting ${placementIds.size} placement records" }
```
Comment (reynders) [inline service/src/main/kotlin/fi/espoo/evaka/varda/VardaUpdateServiceV2.kt:132]: Log the vardaChildId also

File: service/src/main/resources/db/migration/V120__varda_reset_child.sql [hunk 5]
```diff
@@ -0,0 +1,4 @@
+CREATE TABLE varda_reset_child(
+    evaka_child_id UUID primary key NOT NULL,
+    reset_timestamp timestamp with time zone
+);
```
Comment (hruoho) [inline service/src/main/resources/db/migration/V120__varda_reset_child.sql:4]:
```suggestion
);

```

**Pull request merged**